Data/Chimera.hs

```diff
@@ -210,9 +211,34 @@ tabulateM f = Chimera <$> V.generateM (bits + 1) tabulateSubVector
 -- 4862
 -- >>> take 10 (toList ch)
 -- [1,1,2,5,14,42,132,429,1430,4862]
+--
+-- __Note__: Only recursive function calls with decreasing arguments are memoized.
+-- If full memoization is desired, use 'tabulateFix'' instead.
 tabulateFix :: G.Vector v a => ((Word -> a) -> Word -> a) -> Chimera v a
 tabulateFix uf = runIdentity $ tabulateFixM ((pure .) . uf . (runIdentity .))
 
+-- | Fully memoizing version of 'tabulateFix'.
+-- This function will tabulate every recursive call,
+-- but might allocate a lot of memory in doing so.
+-- For example, the following piece of code calculates the
+-- highest number reached by the collatz sequence
+-- of a given number, but also allocates tens of gigabytes of memory,
+-- because the collatz sequence will spike to very high numbers.
```

Comment:
```suggestion
-- because the collatz sequence will spike to very high numbers.
-- because the Collatz sequence will spike to very high numbers.
```

Reply:
I'd add a hyperlink to Wiki about Collatz sequence.